ext/node/ops/crypto/mod.rs

```diff
@@ -1455,3 +1456,102 @@ pub fn op_node_create_private_key(
     _ => Err(type_error("Unsupported algorithm")),
   }
 }
+
+fn parse_public_key(
+  key: &[u8],
+  format: &str,
+  type_: &str,
+) -> Result<pkcs8::Document, AnyError> {
+  match format {
+    "pem" => pkcs8::Document::from_pkcs1_pem(std::str::from_utf8(key).unwrap())
+      .map_err(|_| type_error("Invalid PKCS1 public key")),
+    "der" => {
+      match type_ {
+        "pkcs1" => pkcs8::Document::from_pkcs1_der(key)
+          .map_err(|_| type_error("Invalid PKCS1 public key")),
+        // TODO(@iuioiua): spki type
+        _ => Err(type_error(format!("Unsupported key type: {}", type_))),
+      }
+    }
+    _ => Err(type_error(format!("Unsupported key format: {}", format))),
+  }
+}
+
+#[op2]
+#[serde]
+pub fn op_node_create_public_key(
+  #[buffer] key: &[u8],
+  #[string] format: &str,
+  #[string] type_: &str,
+) -> Result<AsymmetricKeyDetails, AnyError> {
+  let doc = parse_public_key(key, format, type_)?;
+  let pk_info = pkcs8::SubjectPublicKeyInfo::try_from(doc.as_bytes())?;
+
+  let alg = pk_info.algorithm.oid;
+
+  match alg {
+    RSA_ENCRYPTION_OID => {
+      let public_key =
+        rsa::pkcs1::RsaPublicKey::from_der(pk_info.subject_public_key)?;
+      let modulus_length = public_key.modulus.as_bytes().len() * 8;
+
+      Ok(AsymmetricKeyDetails::Rsa {
+        modulus_length,
+        public_exponent: BigInt::from_bytes_be(
+          num_bigint::Sign::Plus,
+          public_key.public_exponent.as_bytes(),
+        )
+        .into(),
+      })
+    }
+    RSASSA_PSS_OID => {
+      let params = PssPrivateKeyParameters::try_from(
```

Comment:
IIUC, the `PssPrivateKeyParameters` should be renamed to `PssKeyParameters`, as it also applies to public keys.